### PR TITLE
refactor(share/availability): don't log error canceled operation

### PR DESCRIPTION
**Pull request by walldiss**
Small cleanup to prevent node from spamming error availability logs on shutdown.

### CODE REVIEW

File: share/availability/full/availability.go [hunk 1]
```diff
@@ -72,7 +72,9 @@ func (fa *ShareAvailability) SharesAvailable(ctx context.Context, root *share.Ro
 
 	_, err := fa.getter.GetEDS(ctx, root)
 	if err != nil {
-		log.Errorw("availability validation failed", "root", root.String(), "err", err.Error())
+		if !errors.Is(err, context.Canceled) {
+			log.Errorw("availability validation failed", "root", root.String(), "err", err.Error())
+		}
```
Comment (Wondertan) [inline share/availability/full/availability.go:77]: Nit: Can return err here

Comment (walldiss) [reply share/availability/full/availability.go:77]: Good one. Thank you

**Review by Wondertan — DISMISSED** [review]

File: share/availability/full/availability.go [hunk 2]
```diff
@@ -72,6 +72,9 @@ func (fa *ShareAvailability) SharesAvailable(ctx context.Context, root *share.Ro
 
 	_, err := fa.getter.GetEDS(ctx, root)
 	if err != nil {
+		if errors.Is(err, context.Canceled) {
```
Comment (renaynay) [inline share/availability/full/availability.go:75]:
```
	_, err := fa.getter.GetEDS(ctx, root)
	if err != nil {
		log.Errorw("availability validation failed", "root", root.String(), "err", err.Error())
		var byzantineErr *byzantine.ErrByzantine
		if ipldFormat.IsNotFound(err) || errors.Is(err, context.DeadlineExceeded) && !errors.As(err, &byzantineErr) {
			return share.ErrNotAvailable
		}
	}
	return err
```

Comment (walldiss) [reply share/availability/full/availability.go:75]: Removed one of the "return err" line

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**